### PR TITLE
Option for s3.end_date_interval and other minor enchancement

### PR DESCRIPTION
**Pull request by gauravmittal1995**
1) Created an option for explicit default in s3.end_date_interval
2) Using S3.start_end_date in hrm_credential

### CODE REVIEW

**Comment by gauravmittal1995** [comment]
@flavour  Made the Changes. Any other Suggestions?
@nursix Please Review

**Comment by gauravmittal1995** [comment]
@flavour Done. Please Review

**Comment by gauravmittal1995** [comment]
@flavour Please Review

**Comment by flavour** [comment]
Very close now :)

File: modules/s3db/hrm.py [hunk 1]
```diff
@@ -6098,7 +6100,17 @@ def prep(r):
...
-    current.response.s3.prep = prep
+    s3.prep = prep
+
+    def postp(r,output):
+        if r.interactive:
+            if not r.component:
+                # Set the minimum end_date to the same as the start_date
+                s3.jquery_ready.append(
+'''S3.start_end_date('hrm_credential_start_date','hrm_credential_end_date')''')
+
```
Comment (gauravmittal1995) [inline modules/s3db/hrm.py]: @flavour I will leave this here for now, so that when @raj454raj does it, he can easily find this and replace it

**Comment by gauravmittal1995** [comment]
@flavour Done, Please Review

**Pull request merged**